sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -202,3 +202,27 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+object Logarithm {
+  def apply(exprs: Seq[Expression]): Expression = {
+    if (exprs.length == 1) {
+      new Log(exprs(0))
+    } else if (exprs.length == 2) {
+      new Logarithm(exprs(0), exprs(1))
+    } else {
+      sys.error(s"Log only accepts two input expressions")
+    }
+  }
+  def apply(child: Expression): Expression = new Log(child)
+}
+
+case class Logarithm(left: Expression, right: Expression)
+  extends BinaryMathExpression((c1, c2) => math.log(c2) / math.log(c1), "LOG") {
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    defineCodeGen(ctx, ev, (c1, c2) => s"java.lang.Math.log($c2) / java.lang.Math.log($c1)") + s"""
```

Comment:
do you mean, when there is only one argument, `Logarithm` generates the codes as the same as `Log`?